src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/ThirdPartyNoticesGenerator.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+public interface IThirdPartyNoticesGenerator
+{
+    Task UpdateThirtPartyNotices();
+}
+
+public class ThirdPartyNoticesGenerator : IThirdPartyNoticesGenerator
+{
+    private static readonly Regex TpnFileName = new(@"third-?party-?notices(.txt)?$", RegexOptions.IgnoreCase);
+
+    private readonly IVmrInfo _vmrInfo;
+    private readonly IVmrDependencyTracker _dependencyTracker;
+    private readonly IFileSystem _fileSystem;
+    private readonly ILogger<ThirdPartyNoticesGenerator> _logger;
+
+    public ThirdPartyNoticesGenerator(
+        IVmrInfo vmrInfo,
+        IVmrDependencyTracker dependencyTracker,
+        IFileSystem fileSystem,
+        ILogger<ThirdPartyNoticesGenerator> logger)
+    {
+        _vmrInfo = vmrInfo;
+        _dependencyTracker = dependencyTracker;
+        _fileSystem = fileSystem;
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// Generates the THIRD-PARTY-NOTICES.txt file by assembling other similar files from the whole VMR.
```

Comment:
```suggestion
    ///     Generates the THIRD-PARTY-NOTICES.txt file by assembling other similar files from the whole VMR.
```

Reply:
I noticed you updated comments in several places... to be compliant with previous changes

Reply:
Yeah, there's this one file that has these indented but honestly I'd prefer to not indent. I only kept that one file in the same shape.